packages/notebook-extension/src/index.ts

```diff
@@ -3560,6 +3561,17 @@ function addCommands(
       return visible;
     }
   });
+
+  // All commands with isEnabled defined directly or in a semantic commands
+  // To simplify here we added all commands as most of them have isEnabled
+  const skip = [CommandIDs.createNew];
+  const notify = () => {
+    Object.values(CommandIDs)
+      .filter(id => !skip.includes(id))
+      .forEach(id => app.commands.notifyCommandChanged(id));
```

Comment:
This seems to be assuming that all commands registered by all the plugins will be available in a lab application.

For example in Notebook the command `notebook:create-output-view` is not registered, leading to an error in the dev tools console: https://github.com/jupyter/notebook/pull/7312#issuecomment-2039087409

Reply:
Shouldn't a plugin take care of notifying the command it registered instead of all commands in the `CommandIDs` namespace?

Reply:
Opened https://github.com/jupyterlab/jupyterlab/issues/16113 so this can be fixed before 4.2.0